BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -0,0 +1,594 @@
+                short* sourceLine = source + y * width;
+                for (int x = 0; x < width; x++)
+                {
+                    depthDataMeters[y, x] = (float)(_depthScale * (short)*sourceLine++);
+                }
+            }
+
+            return depthDataMeters;
+        }
+
+        unsafe private FloatCameraImage CalcIRImage(RealSense2API.RS2Frame frame)
+        {
+            if (!frame.IsValid())
+            {
+                log.Error("IR frame is not valid...\n");
+                return null;
+            }
+
+            int height = DepthResolution.Y;
+            int width = DepthResolution.X;
+
+            FloatCameraImage depthDataMeters = new FloatCameraImage(width, height);
+            byte* source = (byte*)RealSense2API.GetFrameData(frame);
+
+            for (int y = 0; y < height; y++)
+            {
+                byte* sourceLine = source + y * width;
+                for (int x = 0; x < width; x++)
+                {
+                    depthDataMeters[y, x] = (float)((byte)*sourceLine++);
+                }
+            }
+
+            return depthDataMeters;
+        }
+
+        unsafe private ColorCameraImage CalcColor()
+        {
+            if (!_currentColorFrame.IsValid())
+            {
+                log.Error("Color frame is not valid...\n");
+                return null;
+            }
+
+            int height = ColorResolution.Y;
+            int width = ColorResolution.X;
+
+            Bitmap bitmap = new Bitmap(width, height, PixelFormat.Format24bppRgb);
+            Rectangle imageRect = new Rectangle(0, 0, width, height);
+            BitmapData bmpData = bitmap.LockBits(imageRect, ImageLockMode.WriteOnly, bitmap.PixelFormat);
+
+            byte* source = (byte*) RealSense2API.GetFrameData(_currentColorFrame);
+            byte* target = (byte*) (void*)bmpData.Scan0;
+            for (int y = 0; y < height; y++)
+            {
+                byte* sourceLine = source + y * width * 3;
+                for (int x = 0; x < width; x++)
+                {
+                    target[2] = *sourceLine++;
+                    target[1] = *sourceLine++;
+                    target[0] = *sourceLine++;
+                    target += 3;
+                }
+            }
+
+            bitmap.UnlockBits(bmpData);
+            ColorCameraImage image = new ColorCameraImage(bitmap);
+
+            return image;
+        }
+
+        #endregion
+        #endregion
+
+        public void LoadConfigPreset(AdvancedMode.Preset preset)
+        {
+            LoadCustomConfig(AdvancedMode.GetPreset(preset));
+        }
+        
+        public void LoadCustomConfig(string json)
+        {
+            RealSense2API.RS2Device dev = RealSense2API.GetActiveDevice(_pipeline);
+
+            if (!RealSense2API.AdvancedModeEnabled(dev))
+            {
+                RealSense2API.EnabledAdvancedMode(dev, true);
+            }
+
+            RealSense2API.LoadAdvancedConfig(json, dev);
```

Comment:
Could loading a preset change the depth scale? Then it must be reloaded after loading a preset.

Reply:
At least it wouldn't hurt. fixed